types/ember__debug/index.d.ts

```diff
@@ -87,5 +87,13 @@ export function deprecate(
          * An optional url to the transition guide on the emberjs.com website.
          */
         url?: string | undefined;
+        /**
+         * A namespace for the deprecation, usually the package name.
+         */
+        for?: string;
+        /**
+         * Describes when the deprecation became available and enabled.
+         */
+        since?: Partial<Record<'available' | 'enabled', string>>;
```

Comment:
Since the goal is for these types to match what Ember expects, let's make them *actually* match! Specifically, these properties are not optional in the core implementation.

```suggestion
        /**
         * A namespace for the deprecation, usually the package name.
         */
        for: string;
        /**
         * Describes when the deprecation became available and enabled.
         */
        since: Partial<Record<'available' | 'enabled', string>>;
```